### PR TITLE
Merge feature/spring-boot-3 branch to main branch

### PR DESCRIPTION
**Pull request by moarychan**
We plan to stop releasing Spring Cloud Azure 4.x and only release Spring Cloud Azure 5.x in the future, so we will only maintain the main branch for Spring Boot 3 in the future.

Changelogs:
- Merge branch `feature/spring-boot-3` to `main` branch, only keep the source code from `sdk/spring` directory same with `feature/spring-boot-3`.
- Spring models will pick up the external dependency versions of Spring Boot 3 managed, and tagged with a fixed prefix `springboot3_` in `eng/versioning/external_dependencies.txt`.
- Non Spring models will still pick up the external dependency of Spring Boot 2 managed, their dependency versions will be decoupled from the Spring module dependencies.
- Remove cosmos test emulator with java 8.
- Add java 17 test agent for Spring combability test.

### CODE REVIEW

File: sdk/cosmos/ci.spring.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+extends:
```
Comment (saragluna) [inline sdk/cosmos/ci.spring.yml:1]: Moving the running of the azure-spring-data-cosmos to this separate pipeline.

File: sdk/parents/azure-client-sdk-parent/pom.xml [hunk 2]
```diff
@@ -307,29 +307,28 @@
...
-                <artifactId>gummy-bears-api-26</artifactId>
-                <version>0.8.0</version> <!-- {x-version-update;com.toasttab.android:gummy-bears-api-26;external_dependency} -->
-              </signature>
-              <failOnError>false</failOnError>
-              <ignores>${animal.sniffer.ignores}</ignores>
-            </configuration>
-          </execution>
-        </executions>
-      </plugin>
+      <!-- exclude to avoid so many Undefined reference errors -->
```
Comment (saragluna) [inline sdk/parents/azure-client-sdk-parent/pom.xml:310]: Please revert changes in this file

File: eng/versioning/external_dependencies.txt [hunk 3]
```diff
@@ -448,3 +448,98 @@ spring_com.microsoft.azure:azure;1.34.0
 
 # sdk\spring\spring-cloud-azure-service\pom.xml
 spring_redis.clients:jedis;3.8.0
+
+# sdk\spring\pom.xml modules
```
Comment (saragluna) [inline eng/versioning/external_dependencies.txt:452]: please sort these lines

**Review by saragluna — CHANGES_REQUESTED** [review]

**Comment by moarychan** [comment]
> Could you please re-enable `spring-cloud-azure-starter-monitor` and `spring-cloud-azure-starter-monitor-test`, and move them to Spring Boot 3?

Revert the deletion, the two modules are still using the Spring Boot 2.x dependencies.

**Pull request closed without merging**